llvm/lib/CheerpWriter/CheerpWriter.cpp

```diff
@@ -6247,6 +6260,9 @@ void CheerpWriter::compileHelpers()
 		stream << "var " << namegen.getBuiltinName(NameGenerator::Builtin::STACKPTR) << '=' <<
 			linearHelper.getStackStart() << "|0;" << NewLine;
 	}
+
+	stream << "var " <<  EnvironName << "=[],";
+	stream << ArgvName << "=[];" << NewLine;
```

Comment:
I'm not a 100% sure about this, as currently this is always emitted at the start of the file

Reply:
also, does it work when using a closure module? you probably have to use window/this/globalThis in that case to force the variables to be visible outside.

Also, maybe we could only have one variable, with two fields argv and env? but I don't have a good default name for it.

Reply:
it does not work with a closure module or with no modules: you are defining the variables here, but really they should be defined from the outside, in the global scope, and you just use them here (checking that they exist).
I wrote a quick test, and if I just delete these lines altogether, it works. Otherwise, the content is always empty.